src/svg-renderer.js

```diff
@@ -366,32 +398,39 @@ class SvgRenderer {
     }
 
     /**
-     * Draw the SVG to a canvas.
+     * Synchronously draw the loaded SVG to this renderer's `canvas`.
+     * @param {number} [scale] - Optionally, also scale the image by this factor.
+     */
+    draw (scale) {
+        if (!this.loaded) throw new Error('SVG image has not finished loading');
+        this._drawFromImage(scale);
+    }
+
+    /**
+     * Asynchronously draw the (possibly non-loaded) SVG to a canvas.
      * @param {number} [scale] - Optionally, also scale the image by this factor.
      * @param {Function} [onFinish] - An optional callback to call when the draw operation is complete.
+     * @deprecated Use the `loadSVG` and public `draw` method instead.
      */
     _draw (scale, onFinish) {
         // Convert the SVG text to an Image, and then draw it to the canvas.
-        if (this._cachedImage) {
-            this._drawFromImage(scale, onFinish);
+        if (this._cachedImage === null) {
```

Comment:
This one should stay as `this._cachedImage === null` to avoid creating duplicate images